src/lambda/config/templates.ts

```diff
@@ -93,47 +99,178 @@ export async function load(
+        case jsonParser.ParseErrorCode.InvalidUnicode:
             return 'invalid unicode'
-        case ParseErrorCode.PropertyNameExpected:
+        case jsonParser.ParseErrorCode.PropertyNameExpected:
             return 'property name expected'
-        case ParseErrorCode.UnexpectedEndOfComment:
+        case jsonParser.ParseErrorCode.UnexpectedEndOfComment:
             return 'unexpected end of comment'
-        case ParseErrorCode.UnexpectedEndOfNumber:
+        case jsonParser.ParseErrorCode.UnexpectedEndOfNumber:
             return 'unexpected end of number'
-        case ParseErrorCode.UnexpectedEndOfString:
+        case jsonParser.ParseErrorCode.UnexpectedEndOfString:
             return 'unexpected end of string'
-        case ParseErrorCode.ValueExpected:
+        case jsonParser.ParseErrorCode.ValueExpected:
             return 'value expected'
         // By omitting the default case, we force the compiler to yell at us
         // if any enum members are added/removed/changed.
     }
 }
+
+export class TemplatesConfigPopulator {
+    private isDirty: boolean = false
+
+    public constructor(
+        private json: string,
+        private readonly modificationOptions: jsonParser.ModificationOptions = {
+            formattingOptions: {
+                insertSpaces: true,
+                tabSize: 4,
+            },
+        }
+    ) {
+    }
+
+    public ensureTemplateSectionExists(templateRelativePath: string): TemplatesConfigPopulator {
+        this.ensureTemplatesSectionExists()
+
+        this.ensureJsonObjectExists(['templates', templateRelativePath], {})
+
+        return this
+    }
+
+    public ensureTemplateHandlerSectionExists(
+        templateRelativePath: string,
+        handler: string
+    ): TemplatesConfigPopulator {
+        this.ensureTemplateHandlersSectionExists(templateRelativePath)
+
+        this.ensureJsonObjectExists(
+            ['templates', templateRelativePath, 'handlers', handler],
+            {
+                event: {},
+                environmentVariables: {}
+            }
+        )
+
+        return this
+    }
+
+    public ensureTemplateHandlerPropertiesExist(
+        templateRelativePath: string,
+        handler: string
+    ): TemplatesConfigPopulator {
+        this.ensureTemplateHandlerSectionExists(templateRelativePath, handler)
+
+        this.ensureJsonObjectExists(
+            ['templates', templateRelativePath, 'handlers', handler, 'event'],
+            {}
+        )
+
+        this.ensureJsonObjectExists(
+            ['templates', templateRelativePath, 'handlers', handler, 'environmentVariables'],
+            {}
+        )
+
+        return this
+    }
+
+    public getResults(): {
+        isDirty: boolean,
+        json: string,
+    } {
+        return {
+            isDirty: this.isDirty,
+            json: this.json,
+        }
+    }
+
+    private isUnexpectedObjectType(nodeType: jsonParser.NodeType): boolean {
```

Comment:
In JSON, `object`  is not a common base type--i.e. a node can't be both a `string` and an `object`. So you just need to compare `nodeType` with `'object'`.

Reply:
We also want to allow the 'null' type, because we can replace the field in that case.